cogs/groupsCog.py

```diff
@@ -4,288 +4,211 @@
+        group: str = param(description=msg["group_prefix"]),
+        player: discord.Member = param(description=msg["gremove_player"])
     ):
-        group = sanitize_group_name(group_prefix, ctx.author)
-        status, result_str = organize_group.remove_player(sanitize_single(ctx.guild), group, player.id)
-
-        if status:
-            group_role = get(ctx.guild.roles, name=group)
-            # removing group role when existing
-            if group_role in player.roles:
-                await player.remove_roles(group_role)
-                logging.debug(f"Removed role {group_role} from user {player}.")
-            text_channel = ctx.guild.get_channel(organize_group.get_main_channel(sanitize_single(ctx.guild), group))
-            await text_channel.send(msg["gremove_info"].format(player=player.id))
-
+        group = Group.load(group, ctx=ctx)
+        status, result_str = await group.remove_player(player.id)
         await ctx.reply(result_str)
 
-    @commands.command(help=msg["gjoin_help"], aliases=['gbeitreten'])
+    @command(help=msg["gjoin_help"], aliases=['gbeitreten'])
     async def gjoin(self, ctx, 
-        group: str = commands.parameter(description=msg["gjoin_group"])
+        group_name: str = param(description=msg["gjoin_group"])
     ):
-        group = re.sub('[^0-9a-zA-Z\-_]+', '', group.replace(" ", "-")).lower()
-
-        if organize_group.is_owner(sanitize_single(ctx.guild), group, ctx.author.id):
-            status, result_str = False, msg["join_own_group"]
-        else:
-            status, result_str = organize_group.add_self(sanitize_single(ctx.guild), group, ctx.author.id)
-
-        if status:
-            # get the role by group name
-            group_role = get(ctx.guild.roles, name=group)
-            await ctx.author.add_roles(group_role)
-            logging.debug(f"Added role {group_role} to user {ctx.author}.")
-            text_channel = ctx.guild.get_channel(organize_group.get_main_channel(sanitize_single(ctx.guild), group))
-            await text_channel.send(msg["gjoin_info"].format(player=ctx.author.id))
-
-        await ctx.reply(result_str)
-
-    @commands.command(help=msg["gleave_help"], aliases=['gaustreten'])
+        # We could use directly group.add_player() here, the if clauses are only to
+        # provide more detailed response messages.
+        group = Group.load(group_name, ctx=ctx)
+        if group.is_member(ctx.author):
+            await ctx.reply(msg["gjoin_already"])
+            return
+        if len(group.players) >= group.max_players:
+            await ctx.reply(msg["gadd_full"])
+            return
+        if group.is_owner(ctx.author):
+            await ctx.reply(msg["gjoin_owner"])
+            return
+        status, answer = await group.add_player(ctx.author)
+        await ctx.reply(answer)
+
+
+    @command(help=msg["gleave_help"], aliases=['gaustreten'])
     async def gleave(
```

Comment:
I get `argument of type 'NoneType' is not iterable` when there's no group.
The error message for when a group owner wants to leave their own group is now misleading: "This player is not part of the group." - instead they should get a message like "Can't leave your own group."